packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -56,15 +56,14 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
             this.signer = signer;
             storeSignerType(signerType);
           }
-          this.emit('connect', { chainId: hexStringFromNumber(this.signer.chainId) });
-          return this.signer.accounts;
+          this.emit('connect', { chainId: hexStringFromNumber(await this.signer.chainId) });
+          return await this.signer.accounts;
         }
 
         case 'net_version':
-          return this.signer?.chainId ?? 1;
+          return (await this.signer?.chainId) ?? 1;
         case 'eth_chainId':
-          return hexStringFromNumber(this.signer?.chainId ?? 1);
-
+          return hexStringFromNumber((await this.signer?.chainId) ?? 1);
```

Comment:
let's revisit all this

signer.accounts or signer.chainId
they are just arbitrary made-up properties we thought could be useful.

provider.request({method: net_version}) doesn't need to be this complex, if we just relay it directly to signer.

Reply:
yeah, I think we can just remove net_version and eth_chainId from here and those requests will be passed to signer.request instead

Reply:
[c627d48](https://github.com/coinbase/coinbase-wallet-sdk/pull/1353/commits/c627d48c5092f1b30c54815b48e8d2d500b5b934)